filebeat/harvester/harvester.go

```diff
@@ -27,17 +26,16 @@ import (
 )
 
 type Harvester struct {
-	Path               string /* the file path to harvest */
-	Config             harvesterConfig
-	offset             int64
-	State              file.State
-	stateMutex         sync.Mutex
-	SpoolerChan        chan *input.FileEvent
-	encoding           encoding.EncodingFactory
-	file               source.FileSource /* the file being watched */
-	ExcludeLinesRegexp []*regexp.Regexp
-	IncludeLinesRegexp []*regexp.Regexp
-	done               chan struct{}
+	Path            string /* the file path to harvest */
+	Config          harvesterConfig
+	offset          int64
+	State           file.State
+	stateMutex      sync.Mutex
+	SpoolerChan     chan *input.FileEvent
+	encodingFactory encoding.EncodingFactory
+	encoding        encoding.Encoding
```

Comment:
I think it makes sense to just pass the encodingFactory and establish an Encoding instance when required.